Application/Test.Common/MsSqlDatabaseHelper.cs

```diff
@@ -38,6 +41,49 @@ public void CopyDatabase(string originalDatabaseName, string newDatabaseName)
             }
         }
 
+        public string DownloadAndRestoreDatabase()
+        {
+            try
+            {
+                var psScript = @"
+                    $ErrorActionPreference = 'Stop'
+
+                    .\Initialize-PowershellForDevelopment.ps1
+
+                    $settings = @{ 
+                        ApiSettings = @{ Engine = 'SQLServer' } 
+                        ConnectionStrings = @{ EdFi_Ods = '" + _odsConnectionString + @"' } 
+                    }
+                    Set-DeploymentSettings $settings
+
+                    Reset-TestPopulatedTemplateDatabase
+                ";
+
+                var info = new ProcessStartInfo("powershell", psScript)
+                {
+                    WorkingDirectory = Path.GetFullPath("../../../../../../Ed-Fi-ODS-Implementation"),
+                    RedirectStandardError = true,
+                    RedirectStandardOutput = true,
+                };
+
+                using var process = Process.Start(info);
+
+                process.OutputDataReceived += (sender, e) => Console.WriteLine(e.Data);
+                process.BeginOutputReadLine();
+
+                process.ErrorDataReceived += (sender, e) => Console.WriteLine(e.Data);
+                process.BeginErrorReadLine();
+
+                process.WaitForExit();
+            }
+            catch
+            { 
+                return $"Couldn't download template database or restore failed";
```

Comment:
This catch statement isn't executed in case of an error, please check `process.ExitCode != 0` to see if the process failed.

May I suggest throwing an Exception instead of returning an string, perhaps `throw new InvalidOperationException("Couldn't download template database or restore failed");` and catch it in the caller.